examples/azure/terraform-azure-nsg-example/main.tf

```diff
@@ -0,0 +1,118 @@
+  resource_group_name = azurerm_resource_group.main.name
+}
+
+resource "azurerm_subnet" "internal" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.main.name
+  virtual_network_name = azurerm_virtual_network.main.name
+  address_prefixes       = ["10.0.17.0/24"]
+}
+
+resource "azurerm_network_interface" "main" {
+  name                = "${var.prefix}-nic"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+
+  ip_configuration {
+    name                          = "terratestconfiguration1"
+    subnet_id                     = azurerm_subnet.internal.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_network_security_group" "main" {
+  name                = "testSecurityGroup1"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+}
+
+resource "azurerm_network_interface_security_group_association" "main" {
+  network_interface_id      = azurerm_network_interface.main.id
+  network_security_group_id = azurerm_network_security_group.main.id
+}
+
+resource "azurerm_network_security_rule" "main" {
+  name                        = "allowSSH"
+  description                 = "allowSSH"
+  priority                    = 100
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = 22
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  resource_group_name         = azurerm_resource_group.main.name
+  network_security_group_name = azurerm_network_security_group.main.name
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A VIRTUAL MACHINE RUNNING UBUNTU
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_virtual_machine" "main" {
+  name                             = "${var.prefix}-vm"
+  location                         = azurerm_resource_group.main.location
+  resource_group_name              = azurerm_resource_group.main.name
+  network_interface_ids            = [azurerm_network_interface.main.id]
+  vm_size                          = var.vm_size
+  delete_os_disk_on_termination    = true
+  delete_data_disks_on_termination = true
+
+  storage_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "16.04-LTS"
+    version   = "latest"
+  }
+
+  storage_os_disk {
+    name              = "terratestosdisk1"
+    caching           = "ReadWrite"
+    create_option     = "FromImage"
+    managed_disk_type = "Standard_LRS"
+  }
+
+  os_profile {
+    computer_name  = var.hostname
+    admin_username = var.username
+    admin_password = var.password
```

Comment:
See: https://github.com/gruntwork-io/terratest/pull/626/files#r490159868

Reply:
Setting random password from test code;

Reply:
Can you switch to using the `random_password` resource? We should be consistent in how we handle this. See https://github.com/gruntwork-io/terratest/blob/master/examples/azure/terraform-azure-availabilityset-example/main.tf#L109